### DISCUSSION TITLE
composing a curried function type clash 🤨

### DISCUSSION THREAD

**Discussion by sugoidesune** [millsp/ts-toolbelt#205] [Q&A] [answered]
I took these straight from the examples and the curried function inside compose is throwing an error.
The curried function has the correct type.
The composed function works without the curried one.

```ts
declare const compose: Function.Compose
declare function curry<Fn extends Function.Function>(fn: Fn): Function.Curry<Fn>

const a = curry((a1: number, d2:  number) => `${a1 + d2}`)
const c = (c1: string[]) => [c1]
const b = (b1: string) => [b1]

compose(c, b, a)(23,42)
```
You can try it in the ts-toolbelt codesandbox:
https://codesandbox.io/s/ts-toolbelt-x4jly?file=/src/index.ts

The error:
```
const a: Function.Curry<(a1: number, d2: number) => string>

Argument of type 'Curry<(a1: number, d2: number) => string>' is not assignable to parameter of type 
'Function<[a1?: number | x, d2?: number | x], string>'.
  Type 'Curry<(a1: number, d2: number) => string>' is not assignable to type 'string'.ts(2345)
  ```

**Comment by millsp** [comment, marked as answer]
Thanks for reporting this, I have fixed it in the latest release.

**Reply by sugoidesune** [reply]
Thank you for your lightning reply and release.
I tried 9.5.1 and 9.5.0. I also tried to look at what you might have changed in the last release. I still get the error.

I forked the codesandbox with the example and ts-toolbelt set to 9.5.1
https://codesandbox.io/s/ts-toolbelt-951-1pep5?file=/src/index.ts


TS newb here. 😅

**Reply by millsp** [reply]
Not newbie here. It's my bad, sorry. I was doing too many things at the same time. I did push fixes for `Pipe` but not for `Compose`. You can try with `Pipe` so long, but the fix will be out in 15 minutes.

**Reply by millsp** [reply]
Fix is on the way out. Thanks for reporting this!

**Reply by millsp** [reply]
Feel free to ask more questions!

**Reply by sugoidesune** [reply]
Awesome. 9.5.2 works!
Than you! 👏🙏

**Reply by sugoidesune** [reply]
I stumbled on this by using Ramda, would you have any advice on how to fix `R.compose` types in a global fashion.
The only thing I managed was  `const compose: F.compose = R.compose` but it would be great if I could just re-type `R.compose` globally.

**Reply by millsp** [reply]
You will need to send a PR to https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ramda/index.d.ts. Ramda already has `ts-toolbelt` as a dependency. But you will need to upgrade the dependency to version `9` (currently on `6`). There's a bit of work to do that, we could co-author this if you'd like.

**Reply by sugoidesune** [reply]
Yeah saw that, but I started typescript this week  so wasn't gonna try that on my own 😅
Well managed to, Fork, partially clone `types/ramda` and changed the version and install.
Don't know how to run the tests yet and how to check if something broke.

But just a quick look from VsCode errors, A.Clean seems missing.  `O.AssignUp` and `O.MergUp` dropped the "Up" and broke things.

Not sure how much help I will be but just say how you would like to continue with that.

**Reply by sugoidesune** [reply]
Still working through your advanced types medium post though!
